src/main/java/org/opensearch/neuralsearch/processor/combination/ScoreCombinationUtil.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.processor.combination;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Collection of utility methods for score combination technique classes
+ */
+public class ScoreCombinationUtil {
+    private static final String PARAM_NAME_WEIGHTS = "weights";
+
+    /**
+     * Get collection of weights based on user provided config
+     * @param params map of named parameters and their values
+     * @return collection of weights
+     */
+    protected List<Float> getWeights(final Map<String, Object> params) {
```

Comment:
These need to be public, make the class package private.

Reply:
ack